File: Python/is_same_tree.py
# from logging.config import valid_ident
# from turtle import left
from typing import Optional


# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class IsSameTree:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        # print("P VAL: {}" )
        if p == None and q == None:
            return True
        elif p != None or q != None:
            return False
        else:
            if p.val == q.val:
                return self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right)
            else:
                return False


obj = IsSameTree()
t1 = TreeNode(1, TreeNode(2), TreeNode(3))
t2 = TreeNode(1, TreeNode(2), TreeNode(3))

print(obj.isSameTree(t1, t2))

File: Java/IsSameTree.java
import com.sun.source.tree.Tree;

import java.util.Arrays;

public class IsSameTree {
    class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        public TreeNode () {}
        public TreeNode (int val) {
            this.val = val;
        }
        public TreeNode (int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.left = left;
            this.right = right;
        }
    }

    public boolean isSameNode (TreeNode p, TreeNode q) {
        if ((p != null && q == null) || (p == null && q != null)) {
            return false;
        } else if (p != null && q != null) {
            if (p.val == q.val) {
                return isSameNode(p.left, q.left) && isSameNode(p.right, q.right);
            } else {
                return false;
            }
        }

        return true;
    }

    public boolean isSameTree(TreeNode p, TreeNode q) {
        if (isSameNode(p, q)) {
            return true;
        }

        return false;
    }

    public void func() {
        TreeNode p = new TreeNode(1, new TreeNode(2, null, null), new TreeNode( 3, null, null));
        TreeNode q = new TreeNode(1, new TreeNode(2, null, null), new TreeNode( 3, null, null));

        p = new TreeNode(1, new TreeNode(2, null, null), null);
        q = new TreeNode(1, null, new TreeNode(2, null, null));

        System.out.println(isSameTree(p, q));
    }

    public static void main (String[] args) {
        IsSameTree sb = new IsSameTree();
        sb.func();
    }
}
